circuit_knitting/utils/transforms.py

```diff
@@ -158,12 +167,26 @@ def _partition_labels_from_circuit(
     qubit_subsets = connected_components(graph)
     qubit_subsets.sort(key=min)
 
+    # By default, filter qubit_subsets to remove idle wires
+    if not keep_idle_wires:
+        # Determine which qubit wires are idle/unused
+        idle_wires = set(range(circuit.num_qubits))
+        for instruction in circuit.data:
+            for q1 in instruction.qubits:
+                q1_id = circuit.find_bit(q1).index
+                idle_wires.discard(q1_id)
+        # Replace qubit_subsets with filtered list, removing idle qubits
+        qubit_subsets = [
+            subset
+            for subset in qubit_subsets
+            if not (len(subset) == 1 and next(iter(subset)) in idle_wires)
```

Comment:
We have to use `next(iter(subset))` here because that is the standard way to get the "first" element from a `set`.